lib/benchee/benchmark/scenario_context.ex

```diff
@@ -8,6 +8,7 @@ defmodule Benchee.Benchmark.ScenarioContext do
     :current_time,
     :end_time,
     :benchmarking_function,
+    :scenario_input, # before_scenario can alter the original input
```

Comment:
Where do you think would be a good place to document this behavior for folks? It seems to me like this could possibly trip some folks up if they're using `before_scenario`. I _do_ think it's a good feature, but it's one that I think we should thoroughly document - I just can't think of the _right_ place for that documentation to go. Maybe in the top level documentation?

Reply:
Definitely top level documentation imo as in the README.md (and start including that in the hexdocs). My plan is to write a whole section about it and mention these gotchas that the return values of `before_*` will be treated as input.

Also some overview like this:

```
before_scenario_global
before_scenario_local

before_each_global
before_each_local
actual_benchmark
after_each_local
after_each_global

before_each_global
before_each_local
actual_benchmark
after_each_local
after_each_global

# ... 

after_scenario...
```

I wanna write it once all the features are in so that I don't write incomplete docs/docs that are changing

Reply:
Sounds good to me! Good plan on doing in all together.